src/how-to/create-a-github-repository-secret.html

```diff
@@ -0,0 +1,159 @@
+            <a href="https://github.com/p-n-c/website/settings/secrets/actions"
+              >secrets/actions</a
+            >
+          </p>
+          <p>
+            You can now use the repo secret in a GitHub action to grant
+            permissions.
+          </p>
+        </section>
+      </article>
+      <section aria-labelledby="appendix">
+        <h2 id="appendix">Appendix</h2>
+        <p>
+          This appendix was written by the LLM
+          <a href="https://claude.ai/">Claude</a> in response to a request to
+          summarise how repository secrets and
+          <abbr title="Personal Access Tokens">PATs</abbr> are used in GitHub.
+        </p>
+        <section aria-labelledby="pats">
+          <h3 id="pats">Personal Access Tokens (PATs):</h3>
+          <ul>
+            <li>Act as your personal credentials/password for GitHub</li>
+            <li>
+              Allow you to authenticate when using GitHub's API or command line
+            </li>
+            <li>
+              Can be configured with specific permissions (like repo access,
+              workflow management, etc.)
+            </li>
+            <li>Are tied to your personal GitHub account</li>
+            <li>You control their lifespan and can revoke them at any time</li>
+          </ul>
+        </section>
+        <section aria-labelledby="repo-secrets">
+          <h3 id="repo-secrets">Repository Secrets:</h3>
+          <ul>
+            <li>
+              Are encrypted environment variables stored at the repository level
+            </li>
+            <li>
+              Allow you to securely store sensitive data (like PATs, API keys,
+              passwords)
+            </li>
+            <li>Are only accessible within GitHub Actions workflows</li>
+            <li>Cannot be viewed once created, only updated or deleted</li>
+            <li>Are automatically masked in logs if accidentally printed</li>
+          </ul>
+        </section>
+        <section aria-labelledby="how-pats-and-secrets-work-together">
+          <h3 id="how-pats-and-secrets-work-together">
+            How PATs and repository secrets work together:
+          </h3>
+          <ul>
+            <li>You create a PAT with specific permissions</li>
+            <li>You store that PAT as a repository secret</li>
+            <li>
+              Your GitHub Actions workflow references the secret using ${{
+              secrets.SECRET_NAME }}
+            </li>
+            <li>
+              GitHub securely injects the PAT value during workflow execution
+            </li>
+            <li>
+              The action can then use this token to authenticate and perform
+              operations on your behalf
+            </li>
+          </ul>
+        </section>
+        <p>
+          This setup keeps sensitive credentials secure while still allowing
+          automated processes to interact with GitHub.
```

Comment:
I would move this explanation at the beginning, because I got confused: PATs are supposed to be personal, but then you use them to create a secret that can be used by anybody having PR access to the repo.

Reply:
I disagree. I think it's hard to know what the value is until the relationship between the two is understood.

It's one of those times when the developer has to read the documentation closely ;-)